packages/core/src/stateUtils.ts

```diff
@@ -1606,7 +1606,22 @@ export function macrostep(
   }
 
   let nextSnapshot = snapshot;
-  const states: AnyMachineSnapshot[] = [];
+  const microstates: AnyMachineSnapshot[] = [];
+
+  function addMicrostate(
+    microstate: AnyMachineSnapshot,
+    event: AnyEventObject,
+    transitions: AnyTransitionDefinition[]
+  ) {
+    actorScope.system._sendInspectionEvent?.({
+      type: '@xstate.microstep',
+      actorRef: actorScope.self,
+      event,
+      snapshot: microstate,
+      transitions
```

Comment:
Good catch; will revise

Reply:
Actually, to avoid extra processing, I'm going to send the transitions as-is. It is up to the inspector to get what it needs. Unless there's a better reason why we should do processing up-front?